src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -51,6 +51,10 @@ class TestTargetSetup:
   requirements_pex: Pex
   args: Tuple[str, ...]
   input_files_digest: Digest
+  timeout_seconds: Optional[int]
```

Comment:
Right now, IPRs never support timeouts, which is I think why Alex kept the timeout logic inlined in `run_python_test()`. Maybe one day they will, though? 

Either way, calculating the timeout logically belongs in `TestTargetSetup`. `debug_python_test()` will simply ignore the field for now.